matcher/service/routes.py

```diff
@@ -74,10 +81,27 @@ def match():
 
         flask.current_app.logger.debug('Match for configuration has started: {}'.format(config_note_id))
 
-    except openreview.OpenReviewException as error_handle:
-        flask.current_app.logger.error(str(error_handle))
+    except ConfigNoteInterfaceError as error_handle:
+        error_type = str(error_handle)
+        flask.current_app.logger.error(error_type)
+
+        request_status = 404
 
+        # Set status in config note only when the error is not for config note's existence itself
+        if 'Config note not found' not in error_type:
+            set_config_status(openreview_client, config_note_id, 'Error', message=error_type)
```

Comment:
why don't you do?

```
interface.set_status(...)
```

Reply:
Also I'm not sure about this error, if the config note is not found where are you planning to send the error?

I think `ConfigNoteInterface` constructor needs to have a valid config note if this is not found then the endpoint should return a 404 error and don't initiate the thread.

Reply:
1. For cases where interface object instantiation fails (for reasons such as invalid aggregate invitation, etc.), we can not use `interface.set_status(...)`. Please refer to the test `test_routes_invalid_aggregate_invitation` in `test_integration_fairflow.py`.
2. I have made changes for the situation where config note does not exist. Now, a thread is not started in that situation and a 404 is returned.